mono/sgen/sgen-gc.c

```diff
@@ -1346,6 +1346,16 @@ scan_copy_context_for_scan_job (void *worker_data_untyped, ScanJob *job)
 {
 	WorkerData *worker_data = (WorkerData *)worker_data_untyped;
 
+	if (!job->ops) {
```

Comment:
@kumpera This commit was added when I experimented with changing the context on the fly in order to speed up the finishing of the collection, at the cost of leaving unfinished business. I included it though since the worker ops are not fixed. They can be changed to serial when there is only one worker left, or we go through a suspend/resume step (when interrupting the the concurrent mark due to excessive heap growth) after which we change the context to one that also does copying.

While this commit doesn't really have an impact in practice, it conceptually unifies jobs and idle gray stack draining to "random work that needs to be executed with the worker ops, whatever those are at a given point"